packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -642,24 +663,35 @@ class _MonthPickerState extends State<MonthPicker> {
         children: <Widget>[
           new Semantics(
             sortKey: _MonthPickerSortKey.calendar,
-            child: new PageView.builder(
-              key: new ValueKey<DateTime>(widget.selectedDate),
-              controller: _dayPickerController,
-              scrollDirection: Axis.horizontal,
-              itemCount: _monthDelta(widget.firstDate, widget.lastDate) + 1,
-              itemBuilder: _buildItems,
-              onPageChanged: _handleMonthPageChanged,
+            child: new NotificationListener<ScrollStartNotification>(
+              onNotification: (_) => _handleScrollStartNotification(context),
```

Comment:
You don't need to pass the context along here, since it's a State property. Rather than creating a closure, you could just make the _handleFoo methods NotificationListenerCallbacks

Reply:
Will move the functions into inline closures as they're short and don't clutter the build() code too much.